packages/@aws-cdk-testing/cli-integ/resources/integ.jest.config.js

```diff
@@ -19,9 +20,42 @@ module.exports = {
   // for the lock. Which is almost never what we actually care about. Set it high.
   testTimeout: 2 * 60 * 60_000,
 
-  maxWorkers: 50,
+  maxWorkers: maxWorkers(),
   reporters: [
     "default",
     ["jest-junit", { suiteName: "jest tests", outputDirectory: "coverage" }]
   ]
 };
+
+/**
+ * Based on the machine spec, calcluate the maximum number of jest workers we can start in parallel.
+ */
+function maxWorkers() {
+
+  const totalMemoryMB = os.totalmem() / 1024 / 1024;
+  const totalCores = os.cpus().length;
+
+  // empirically observed. this includes:
+  // - 150 jest test process
+  // - 140 app synthesis subprocess  
+  // - 200 cli subprocess
+  const observedWorkerMemoryMB = 500;
+
+  // GC occasionally gets a chance to clear this amount of memory
+  // so the workers memory doesnt need to be so tight.
+  const estimatedWorkerFluctuationMB = 150;
+
+  // leave 3GB for the OS and other external processes
+  const reservedMemoryMB = 3000;
+
+  // our processes don't take up much CPU so we allow for a large factor.
+  const cpuScaleFactor = 10;
```

Comment:
I'm tempted to leave this out altogether?

Reply:
We probably can...but it doesn't jive well with me saying we are completely not bound by CPU (thinking about has calculation and regex matching). I've increased to 15.